### PR TITLE
Specs cleanup #172770622

### PR DESCRIPTION
**Pull request by riccardo-giomi**
## PT Story
Some specs are failing when rspecs are run randomly.

* data (preconditions) is not being set up correctly

AND/OR

* data is still being left over from previous test being run

#### PT URL:

https://www.pivotaltracker.com/story/show/172770622

## Changes proposed in this pull request:
1. make the specs work when run in random order

### CODE REVIEW

File: spec/db/shared_specs_db_seeding.rb [hunk 1]
```diff
@@ -150,7 +150,7 @@
 #========================================================================================
 
 
-RSpec.shared_examples 'it calls geocode min max times with csv file' do |num_users, geocode_min, geocode_max, csv_filename|
+RSpec.shared_examples 'it calls geocode min max times with csv file' do |num_users, admin_email, admin_pwd, geocode_min, geocode_max, csv_filename|
```
[the remark this answers is not in the document]
Comment (riccardo-giomi) [reply spec/db/shared_specs_db_seeding.rb:153]: Now I see where that bug came from... Bad hound!

[the remark this answers is not in the document]
Comment (aonomike) [reply spec/db/shared_specs_db_seeding.rb:153]: lol

**Comment by riccardo-giomi** [comment]
I'm following the idea of having a .env.test.example that can be renamed to .env.test by a newbie in a new installation, and make the specs run green immediately.

There is nothing in the .env.test.example file that I didn't find in the tests (cassettes, mocks, etc.) so if there is a problem with sensitive data we should change the mocks.

If this is ok I can change the installation and contributing documents to reflect this change.
I would also suggest we do something similar for .env.development.example.

**Comment by weedySeaDragon** [comment]
FYI - I'm looking at the tests and trying to narrow down where some of the problems are (for seed 2889 at least).  Seems that many of the problems are caused by `spec/models/user_spec.rb : Scopes:expiration_dates` (before the changes made in the latest commit).  
I'll post more here if or when I figure out more.

**Comment by riccardo-giomi** [comment]
@weedySeaDragon:
> FYI - I'm looking at the tests and trying to narrow down where some of the problems are (for seed 2889 at least). Seems that many of the problems are caused by `spec/models/user_spec.rb : Scopes:expiration_dates` (before the changes made in the latest commit).
> I'll post more here if or when I figure out more


Tests from seed 2889 should be fine after the last commits. Do they still fail for you?

The last failing run in the CI was with seed 62154 (I'm using the CI to suggest me new seeds to fix).

**Comment by weedySeaDragon** [comment]
They definitely all pass.  But I think that if `spec/models/user_spec.rb : Scopes:expiration_dates` is cleaned up to not use the `before(:all)` to set up all the data at once, some of those other changes aren't really needed.

**Comment by riccardo-giomi** [comment]
seeds 2899, 13584 and 44191 are now ok, next one is 60800.

**Review by weedySeaDragon — APPROVED** [review]

**Pull request merged**